### PR TITLE
MINOR: "partition" typos and method doc arg fix

### PR DESCRIPTION
**Pull request by lbradstreet**
Some minor typos that have annoyed me along with a fix to a docstring.

### CODE REVIEW

File: core/src/main/scala/kafka/log/LogCleanerManager.scala [hunk 1]
```diff
@@ -359,9 +359,10 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
    *
    * @param dataDir                       The File object to be updated
    * @param partitionToUpdateOrAdd        The [TopicPartition, Long] map data to be updated. pass "none" if doing remove, not add
-   * @param topicPartitionToBeRemoved     The TopicPartition to be removed
+   * @param partitionToRemove             The TopicPartition to be removed
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleanerManager.scala:362]: Now that you are modifying this, probably it's worth mentioning that this `Option` parameter has a default value and what it means when `None` is passed.

**Review by jlprat — APPROVED** [review]
Thanks for the PR! LGTM, I only have a small nit about the scaladoc provided.

**Review by showuon — APPROVED** [review]
Nice find! LGTM! Thank you.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**